platform-sdk/docs/proposals/consensus-layer/Consensus-Layer.md

```diff
@@ -326,29 +330,39 @@ We will pass the full round info (metadata + events) to all listeners, and they
 #### Roster Changes
 
 When the roster changes, the Hashgraph algorithm must be made aware. It needs roster information to be able to come
-to consensus. While several modules must respond to roster changes, it is the Hashgraph module that "owns" the roster.
-Each round has an associated roster -- the roster that was used to come to consensus on that round. While the
-Hashgraph owns the roster, it does not _determine_ the roster, this happens in the Execution layer.
+to consensus. While several modules must respond to roster changes, it is the Hashgraph module that determines the
+applicable roster for any given round.
 
 At runtime, there is a service in Execution that exposes to the world API for adding nodes to, or removing nodes from,
 the address book. At some point in time, Execution will decide the time is right to construct a new consensus roster,
 and pass that roster to the consensus system. **This must be deterministic**. Each node in the network must assign the
 same roster to the same rounds. However, Consensus and Execution run in completely different threads, and possibly,
 in different processes.
 
-With this architecture, we introduce a new concept into `EventCore`. `EventCore` has a `payload`, which is a `byte[]`
-that the Execution layer can use to carry whatever payload is meaningful to it. The consensus layer needs an analogous
-section of the `EventCore` into which messages particular to Consensus can be placed. This section will be called
-`repeated control_messages`. A `ControlMessage` is a protobuf message used to control, in some way, the consensus
-system.
-
-TODO: I'm seriously thinking of having "system_transactions" and "user_transactions" instead of "control_messages" and
-"payload".
-
-`ControlMessage` has a one-of, among which is `RosterChangeMessage`. This message will have the new roster, and be
-signed by the network as a whole. Each honest node will submit a `RosterChangeMessage`, and the Hashgraph module will
-recognize these messages. When it encounters the **first** such message for a given roster in a round that has come to
-consensus, the very next round will use that new activated roster.
+Each `EventCore` will have the ability to carry two different types of transactions: `user_transactions` and
+`system_transactions`. A "system" transaction is a transaction known by Consensus, but not Execution, while "user"
+transactions are those supplied by Execution, and unknown to Consensus. By separating these two fields, we make it very
+inexpensive for logic in Consensus to iterate over and handle "system" transactions regardless of the number of user
+transactions in the event.
+
+Each node, individually, decides deterministically at some round that it is time to adopt a new roster, and
```

Comment:
Let's discuss this. It doesn't quite fit with how I was understanding your proposal.